libs/ngxtension/create-injectable/src/create-injectable.ts

```diff
@@ -0,0 +1,32 @@
+import {
+	InjectionToken,
+	type EnvironmentProviders,
+	type Provider,
+} from '@angular/core';
+import {
+	createInjectionToken,
+	type CreateInjectionTokenDeps,
+	type CreateInjectionTokenOptions,
+} from 'ngxtension/create-injection-token';
+
+export type CreateInjectableOptions<
+	TFactory extends (...args: any[]) => object,
+	TFactoryDeps extends Parameters<TFactory> = Parameters<TFactory>,
+> = (TFactoryDeps[0] extends undefined
+	? { deps?: never }
+	: { deps: CreateInjectionTokenDeps<TFactory, TFactoryDeps> }) & {
+	isRoot?: boolean;
+	token?: InjectionToken<ReturnType<TFactory>>;
+	extraProviders?: Provider | EnvironmentProviders;
+};
+
+export function createInjectable<
+	TFactory extends (...args: any[]) => object,
+	TFactoryDeps extends Parameters<TFactory> = Parameters<TFactory>,
+	TOptions extends CreateInjectionTokenOptions<
+		TFactory,
+		TFactoryDeps
+	> = CreateInjectableOptions<TFactory, TFactoryDeps>,
+>(factory: TFactory, options?: TOptions): InjectionToken<ReturnType<TFactory>> {
+	return createInjectionToken(factory, options)[2];
```

Comment:
this won't work because `[2]` is just an `InjectionToken`. It cannot be used on its own in the `providers` array. We'd need a different implementation or adjust `createInjectionToken` somehow

Reply:
I should've added a test for provide, will work on that thanks

Reply:
For now I've just changed it to return the provide function as well so that we have something working, but perhaps we want to leave the exact implementation open to discussion still

Reply:
Sure thing.

Reply:
Pushed my implementation. If you don't like it, we can revert.

Reply:
Looks good to me - I'll update the docs shortly. Do we want to leave the PR open and I can invite people from the video to comment/discuss? or we can just leave it as experimental for now

Reply:
I've updated docs. I think we can merge it as-is. If feedbacks start coming in, we can always re-iterate

Reply:
Adjusted tests to showcase the behavior or root service vs non-root service

Reply:
Sounds good! Nice work!